---
uint8 EMPTY = 0
uint8 VALID = 1

uint8 isvalid
# the next available push plan if any are available
PushPlan plan
